MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -214,10 +221,46 @@ open class RouteProgress: NSObject {
 
                     congestionTimesPerStep[legIndex].append(stepCongestionValues)
                 }
+
+                if let segmentMaximumSpeedLimits = leg.segmentMaximumSpeedLimits {
+                    guard nextManeuverCoordinateIndex < segmentMaximumSpeedLimits.count else { continue }
+
+                    let stepSegmentMaximumSpeedLimits = Array(segmentMaximumSpeedLimits[maneuverCoordinateIndex..<nextManeuverCoordinateIndex])
+                    maximumSpeedLimitsByStep.append(stepSegmentMaximumSpeedLimits)
```

Comment:
Is it necessary to build up this array at all? Seems like ~~`speedLimitDistancesByLeg`~~ `positionedSpeedLimitsByStep` is the more useful property, since that’s what `currentSpeedLimit` uses.

Reply:
Removed.